[libc++][hardening] In production hardening modes, trap rather than abort (#78561)
llvm/llvm-project: libcxx/docs/BuildingLibcxx.rst

.. _assertion-handler:

Overriding the default assertion handler
========================================

When the library wants to terminate due to a hardening assertion failure, the
program is aborted by invoking a trap instruction (or in debug mode, by
a special verbose termination function that prints an error message and calls

> ``std::abort()``). However, vendors can also override that mechanism at CMake
> configuration time.

Reviewer: ```suggestion
``std::abort()``). This is done to minimize the code size impact of enabling hardening in the library. However, vendors can also override that mechanism at CMake
configuration time.
```

> When a hardening assertion fails, the library invokes the
> ``_LIBCPP_ASSERTION_HANDLER`` macro. A vendor may provide a header that contains
> a custom definition of this macro and specify the path to the header via the

Reviewer: ```suggestion
Under the hood, a hardening assertion will invoke the ``_LIBCPP_ASSERTION_HANDLER`` macro upon failure.
A vendor may provide a header that contains
a custom definition of this macro and specify the path to the header via the
```
